experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -255,54 +274,43 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 	invert: (
 		{ revision, change }: TaggedChange<OptionalChangeset>,
 		invertChild: NodeChangeInverter,
+		genId: IdAllocator,
 	): OptionalChangeset => {
-		// Changes to the child that existed in this field before `change` was applied.
-		let originalChildChanges: NodeChangeset | undefined;
-		const inverseChildChanges = new ChildChangeMap<NodeChangeset>();
-		if (change.childChanges !== undefined) {
-			for (const [id, childChange] of change.childChanges) {
-				if (id === "self" && change.fieldChange !== undefined) {
-					originalChildChanges = invertChild(childChange);
-				} else {
-					inverseChildChanges.set(
-						// This makes assumptions about how sandwich rebasing works
-						id,
-						invertChild(childChange),
-					);
-				}
+		const { moves, childChanges } = change;
+		const invertIdMap = new ChildChangeMap<RegisterId>();
+		const withIntention = (id: RegisterId): RegisterId => {
+			if (id === "self") {
+				return id;
 			}
-		}
-
-		const selfChanges = change.fieldChange?.newContent?.changes;
-		if (selfChanges !== undefined) {
-			inverseChildChanges.set("self", invertChild(selfChanges));
-		}
-
-		const inverse: OptionalChangeset = {
-			childChanges:
-				inverseChildChanges.size > 0
-					? Array.from(inverseChildChanges.entries())
-					: undefined,
+			return { revision: id.revision ?? revision, localId: id.localId };
 		};
+		for (const [src, dst] of moves) {
+			invertIdMap.set(dst, src);
+		}
 
-		const { fieldChange } = change;
-		if (fieldChange !== undefined) {
-			// `change` replaces the node in the field
-			inverse.fieldChange = {
-				id: fieldChange.id,
-				wasEmpty: fieldChange.newContent === undefined,
-			};
-
-			if (!fieldChange.wasEmpty) {
-				assert(revision !== undefined, 0x592 /* Unable to revert to undefined revision */);
-				inverse.fieldChange.newContent = {
-					revert: { revision, localId: fieldChange.id },
-					changes: originalChildChanges,
-				};
+		const rebasedMoves: typeof change.moves = [];
+		for (const [src, dst] of moves) {
+			// TODO:AB#6298: This assert can legitimately fail for transactions, meaning we have litte test coverage there.
+			assert(
+				src === "self" || dst === "self",
+				"Only inverses for single-register attached sets are supported.",
+			);
+			if (dst === "self" && src !== "self") {
```

Comment:
Sorry, this is really nitpicky but I keep reading this wrong (it doesn't help that it lines up with the assert above).

```suggestion
			if (src !== "self" && dst === "self") {
```